src/sdk/client.ts

```diff
@@ -67,28 +67,22 @@ export const makeRequestFactory = (
         return { response, options };
       });
 
-      if (!response.ok) {
-        const text = await response.text();
+      const text = await response.text();
+      const body = guard(
+        () => JSON.parse(text),
+        (err) => err instanceof SyntaxError,
+      ) ?? { body: text };
 
+      if (!response.ok) {
         throw new IntegrationError('HTTP error in client', {
           type: 'http',
-          body:
-            guard(
-              () => JSON.parse(text),
-              (err) => err instanceof SyntaxError,
-            ) ?? text,
+          body,
```

Comment:
just want to make sure you know, this is now `body: { body: text }` in the case that it's a text response. Not sure if that's intentional

Reply:
Yeah, I wasn't sure how we were using this — @zlalvani, is it safe to have it this way?